files/lib/system/exporter/WBB3xExporter.class.php

```diff
@@ -3139,22 +3139,28 @@ public function exportCalendarEventDateParticipation($offset, $limit)
      */
     private function countAttachments($type)
     {
+        $packageID = $this->getPackageIdByAttachmentType($type);
+        if (!$packageID) {
+            return 0;
+        }
+
         if (\substr($this->getPackageVersion('com.woltlab.wcf'), 0, 3) == '1.1') {
-            $sql = "SELECT  COUNT(*) AS count
+            $sql = "SELECT  MAX(attachmentID)
                     FROM    wcf" . $this->dbNo . "_attachment
-                    WHERE   containerType = ?
-                        AND containerID > ?";
+                    WHERE   packageID = ?
+                        AND containerID > ?
+                        AND containerType = ?";
         } else {
-            $sql = "SELECT  COUNT(*) AS count
+            $sql = "SELECT  MAX(attachmentID)
                     FROM    wcf" . $this->dbNo . "_attachment
-                    WHERE   messageType = ?
-                        AND messageID > ?";
+                    WHERE   packageID = ?
+                        AND messageID > ?
+                        AND messageType = ?";
         }
         $statement = $this->database->prepareStatement($sql);
-        $statement->execute([$type, 0]);
-        $row = $statement->fetchArray();
+        $statement->execute([$packageID, 0, $type]);
 
-        return $row['count'];
+        return $statement->fetchSingleColumn() ?: 0;
```

Comment:
I'd simplify this to a simple `MAX(attachmentID)` without any conditions. If all features of the source community are used then it's likely that all the maximum IDs of the various containers are roughly equally large. Thus this won't save any iterations – but the counting query is much more complex.